reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -4531,6 +4531,38 @@ public int getPrefetch() {
 	}
 
 	/**
+	 * Keep information about the order in which source values were received by
+	 * indexing them with a 0-based incrementing long, returning a {@link Flux}
+	 * of {@link Tuple2 Tuple2<(index, value)>}.
+	 *
+	 * @return an indexed {@link Flux} with each source value combined with its 0-based index.
+	 */
+	public final Flux<Tuple2<Long, T>> indexed() {
+		return indexed(Tuples::of);
```

Comment:
this could be a static bifunction to avoid :: creation

Reply:
@smaldini a static `BiFunction` will wrap the simple method call into a runtime-generated class for the lambda, adding a method call to the stack. The method reference is directly linked to the original method, adding 0 overhead. I don't think there's a situation where a method reference has more overhead than a lambda (although the difference is minimal, one method call, so favoring a lambda for other reasons like clarity of code could sometimes make sense 😉 )